src/js/game/Entities/BaseEntity.js

```diff
@@ -597,4 +598,33 @@ module.exports = class BaseEntity {
         // no lava or water
         && (groundBlock.blockType !== "water" && groundBlock.blockType !== "lava");
   }
+
+  handleMoveOffPressurePlate(moveOffset) {
```

Comment:
This looks great!  Can we now call `handleMoveOffPressurePlate` and `handleMoveOnPressurePlate` from `BaseEntity.prototype.doMoveForward`?

Reply:
Would you like controller.execute to be refactored to pass along moveOffset? Since entity.handleMoveOffPressurePlate(moveOffset); takes it as an argument, and it never makes it to the entity.doMoveForward().

if we're going to pass along that data, it would have to make its way into controller.execute... and then into entity.addCommand()... I'm not sure the best way to retain that data without making this -really- overly arduous.

Will continue looking into it 👍

Reply:
Update:

The information is captured in baseEntity actually, but the one problem is that doMoveForward() behaves a bit differently depending on function call. 

Basically: If calling moveForward / moveBackward, when we hit doMoveForward() in baseEntity, movementState will be -1, and facing direction will be 0-3. So we'd use facing direction to get offset.

if we call moveDirection() however, facing direction will be wrong. It will be the way the entity WAS facing before the move call and movementState will be 0-3, so that is what we'd use to get offset.

problem is by the time we hit baseEntity in doMoveForward, I don't think we capture what command that call is coming from? This isn't a problem for the Agent because it can't call moveDirection, but the Player causes an issue... We can assume always moveDirection() for this year, but it wouldn't work correctly for past years... not that past years have pressure plates... but it feels bad.